domain/storage/state/storagepool.go

```diff
@@ -478,29 +486,95 @@ ORDER BY sp.uuid`, spTypes, storagePool{}, poolAttribute{})
 	return dbRows.toStoragePools(keyValues)
 }
 
-// GetStoragePoolByName returns the storage pool with the specified name.
+// GetStoragePoolUUID returns the UUID of the storage pool for the specified name.
 // The following errors can be expected:
 // - [storageerrors.PoolNotFoundError] if a pool with the specified name does not exist.
-func (st State) GetStoragePoolByName(ctx context.Context, name string) (domainstorage.StoragePool, error) {
+// Exported for use by other domains that need to load storage pools.
+func GetStoragePoolUUID(
```

Comment:
`GetStoragePoolUUID` should take a `domain.Preparer`, see `github.com/juju/juju/domain/sequence/state.NextValue`

Reply:
Can we please move this to another file, with its own test suite.